channel-email/src/main/java/com/synopsys/integration/alert/channel/email/environment/EmailEnvironmentVariableHandler.java

```diff
@@ -0,0 +1,202 @@
+
+    @Autowired
+    public EmailEnvironmentVariableHandler(EmailGlobalConfigAccessor configAccessor, EnvironmentVariableUtility environmentVariableUtility, EmailGlobalConfigurationValidator validator) {
+        super(ChannelKeys.EMAIL.getDisplayName(), Stream.concat(EMAIL_CONFIGURATION_KEYSET.stream(), OLD_ADDITIONAL_PROPERTY_KEYSET.stream()).collect(Collectors.toSet()));
+        this.configAccessor = configAccessor;
+        this.environmentVariableUtility = environmentVariableUtility;
+        this.validator = validator;
+    }
+
+    @Override
+    protected Boolean configurationMissingCheck() {
+        return !configAccessor.doesConfigurationExist();
+    }
+
+    @Override
+    protected EmailGlobalConfigModel configureModel() {
+        EmailGlobalConfigModel configModel = new EmailGlobalConfigModel();
+        configModel.setName(AlertRestConstants.DEFAULT_CONFIGURATION_NAME);
+        configureEmailSettings(configModel);
+        configureAdditionalProperties(configModel);
+
+        return configModel;
+    }
+
+    @Override
+    protected ValidationResponseModel validateConfiguration(EmailGlobalConfigModel configModel) {
+        return validator.validate(configModel);
+    }
+
+    @Override
+    protected EnvironmentProcessingResult buildProcessingResult(EmailGlobalConfigModel configModel) {
+        EmailGlobalConfigModel obfuscatedModel = configModel.obfuscate();
+
+        EnvironmentProcessingResult.Builder builder = new EnvironmentProcessingResult.Builder(EMAIL_CONFIGURATION_KEYSET)
+            .addVariableNames(OLD_ADDITIONAL_PROPERTY_KEYSET);
+
+        for (String additionalPropertyName : OLD_ADDITIONAL_PROPERTY_KEYSET) {
+            if (environmentVariableUtility.hasEnvironmentValue(additionalPropertyName)) {
+                String value = environmentVariableUtility.getEnvironmentValue(additionalPropertyName).orElse(null);
+                builder.addVariableValue(additionalPropertyName, value);
+            }
+        }
+
+        obfuscatedModel.getSmtpHost().ifPresent(value -> builder.addVariableValue(EMAIL_HOST_KEY, value));
+        obfuscatedModel.getSmtpPort().map(String::valueOf).ifPresent(value -> builder.addVariableValue(EMAIL_PORT_KEY, value));
+        obfuscatedModel.getSmtpFrom().ifPresent(value -> builder.addVariableValue(EMAIL_FROM_KEY, value));
+        obfuscatedModel.getSmtpAuth().map(String::valueOf).ifPresent(value -> builder.addVariableValue(AUTH_REQUIRED_KEY, value));
+        obfuscatedModel.getSmtpUsername().ifPresent(value -> builder.addVariableValue(AUTH_USER_KEY, value));
+
+        if (Boolean.TRUE.equals(obfuscatedModel.getIsSmtpPasswordSet())) {
+            builder.addVariableValue(AUTH_PASSWORD_KEY, AlertConstants.MASKED_VALUE);
+        }
+
+        return builder.build();
+    }
+
+    @Override
+    protected void saveConfiguration(EmailGlobalConfigModel configModel, EnvironmentProcessingResult processingResult) {
+        if (processingResult.hasValues()) {
```

Comment:
Could this always be the case? If so it may be worth pulling it up into the parent class

Reply:
Jira has it slightly different here because it also checks to make sure the configuration by name doesn't exist. We could pull out the first part for the hasValues, but keep the second part for Jira

Reply:
Ya it certainly looks like we can pull the first part out judging by the other handlers